src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -578,19 +578,26 @@ uct_ib_mlx5_verbs_srq_init(uct_ib_mlx5_srq_t *srq, struct ibv_srq *verbs_srq,
 {
     uct_ib_mlx5dv_srq_t srq_info = {};
     uct_ib_mlx5dv_t obj          = {};
-    struct mlx5dv_obj dv         = {};
-    struct mlx5dv_srq dvsrq      = {};
     ucs_status_t status;
     uint16_t stride;
 
-    obj.dv.srq.in  = verbs_srq;
-    obj.dv.srq.out = &srq_info.dv;
+    obj.dv.srq.in         = verbs_srq;
+    obj.dv.srq.out        = &srq_info.dv;
+#if HAVE_DEVX && HAVE_DECL_MLX5DV_SRQ_MASK_SRQN
```

Comment:
why need HAVE_DEVX ?
can just test HAVE_DECL_MLX5DV_SRQ_MASK_SRQN

Reply:
or just HAVE_DEVX

Reply:
@Artemy-Mellanox are you agree to use introduced `HAVE_DECL_MLX5DV_SRQ_MASK_SRQN` instead?

Reply:
redundant though will do the job, HAVE_DECL_MLX5DV_SRQ_MASK_SRQN == HAVE_DEVX

Reply:
> redundant though will do the job, HAVE_DECL_MLX5DV_SRQ_MASK_SRQN == HAVE_DEVX

ok, I think it's better to not define `HAVE_DECL_MLX5DV_SRQ_MASK_SRQN` and check for `HAVE_DEVX` here